module_utils/network_lsr/argument_validator.py

```diff
@@ -1974,6 +1974,12 @@ def _ipv6_is_not_configured(connection):
                     "Setting 'dns_search' or 'dns_options' is not allowed when "
                     "both IPv4 and IPv6 are disabled.",
                 )
+            elif not _ipv4_enabled(connection) and _ipv6_is_not_configured(connection):
```

Comment:
What is the difference between enabled and configured? Can be IPv6 enabled but not configured?

Reply:
> What is the difference between enabled and configured?

In the current network role, if using nm provider as the backend, the role will configure ipv4.method as 'auto' or 'manual' or 'disabled'.  `not _ipv4_enabled` corresponds to the ipv4.method 'disabled'.

As a comparison, the role will configure ipv6.method as  'auto' or 'manual' or 'disabled' or 'ignore'. `_ipv6_is_not_configured` corresponds to the ipv6.method 'ignore'.
 



> Can be IPv6 enabled but not configured? 

No. You can take a look at the code here.  https://github.com/linux-system-roles/network/blob/main/module_utils/network_lsr/argument_validator.py#L703

Reply:
This make sense, thanks for explaining!